src/Tribe/Utils/Post_Thumbnail.php

```diff
@@ -155,44 +157,58 @@ public function fetch_data() {
 			return $this->data;
 		}
 
-		$post_id      = $this->post_id;
 		$image_sizes  = $this->get_image_sizes();
 		$thumbnail_id = $this->thumbnail_id;
 
-		$thumbnail_data = array_combine(
-			$image_sizes,
-			array_map(
-				static function ( $size ) use ( $thumbnail_id ) {
-					$size_data = wp_get_attachment_image_src( $thumbnail_id, $size );
+		$cache_key = empty( $thumbnail_id ) ? -1 : $thumbnail_id;
+
+		if ( empty( $cache_thumbnail[ $cache_key ] ) ) {
+			$thumbnail_data = array_combine(
+				$image_sizes,
+				array_map(
+					static function( $size ) use ( $thumbnail_id ) {
+						static $cache_size_data = [];
+
+						$size_data_cache_key = empty( $thumbnail_id ) ? -1 : $thumbnail_id;
+						$size_data_cache_key = "{$size_data_cache_key}:{$size}";
+
+						if ( ! isset( $cache_size_data[ $size_data_cache_key ] ) ) {
+							$cache_size_data[ $size_data_cache_key ] = wp_get_attachment_image_src( $thumbnail_id, $size );
+						}
+
+						$size_data = $cache_size_data[ $size_data_cache_key ];
+
+						if ( false === $size_data ) {
+							return (object) [
```

Comment:
:no_entry_sign: Short array syntax is not allowed

<sub>hash: 13ef4260ec38e3e969c3a78e17219818</sub>